app/src/main/res/values/strings.xml

```diff
@@ -517,4 +516,85 @@ Otherwise, you can download another keyboard in the app store. Popular keyboards
+    <string name="quest_buildingType_church">Church</string>
+    <string name="quest_buildingType_chapel">Chapel</string>
+    <string name="quest_buildingType_cathedral">Cathedral</string>
+    <string name="quest_buildingType_mosque">Mosque</string>
+    <string name="quest_buildingType_temple">Temple</string>
+    <string name="quest_buildingType_synagogue">Synagogue</string>
+
+    <string name="quest_buildingType_civic">Civic building</string>
+    <string name="quest_buildingType_school">School</string>
+    <string name="quest_buildingType_college">College</string>
+    <string name="quest_buildingType_hospital">Hospital</string>
+    <string name="quest_buildingType_public">Public building</string>
+    <string name="quest_buildingType_public_description">accessible to general public, e.g. town hall or court house</string>
+    <string name="quest_buildingType_stadium">Stadium</string>
+    <string name="quest_buildingType_train_station">Train station</string>
+    <string name="quest_buildingType_transportation">Public transportation building</string>
+    <string name="quest_buildingType_university">University</string>
+
+    <string name="quest_buildingType_cars">For cars</string>
+    <string name="quest_buildingType_carport">Carport</string>
+    <string name="quest_buildingType_carport_description">covered structure used to protect vehicles from rain and snow, mostly open on two or more sides</string>
+    <string name="quest_buildingType_garage">Single garage</string>
+    <string name="quest_buildingType_garages">Multiple garages</string>
+    <string name="quest_buildingType_parking">Parking</string>
+
+    <string name="quest_buildingType_farm">Farm</string>
+    <string name="quest_buildingType_farmhouse">Farmhouse</string>
+    <string name="quest_buildingType_farmhouse_description">main building of a farm</string>
+    <string name="quest_buildingType_farm_auxiliary">Other farm building</string>
+    <string name="quest_buildingType_farm_auxiliary_description">any building on a farm that is not a residential building</string>
+    <string name="quest_buildingType_barn">Barn</string>
+    <string name="quest_buildingType_greenhouse">Greenhouse</string>
+
+    <string name="quest_buildingType_other">Other</string>
+    <string name="quest_buildingType_shed">Shed</string>
+    <string name="quest_buildingType_hut">Hut</string>
+    <string name="quest_buildingType_hut_description">small and crude shelter</string>
+    <string name="quest_buildingType_roof">Roof</string>
+    <string name="quest_buildingType_roof_description">open on at least two sides, e.g. roof of a petrol station</string>
+    <string name="quest_buildingType_cabin">Cabin</string>
+    <string name="quest_buildingType_cabin_description">small, roughly built house typically found in rural areas</string>
+    <string name="quest_buildingType_construction">Construction</string>
+    <string name="quest_buildingType_ruins">Ruins</string>
```

Comment:
note that building=ruins should be applied only to be building constructed as ruins (for example in a https://en.wikipedia.org/wiki/English_landscape_garden ).

Given likely misuse and rarity of this type of [folly](https://en.wikipedia.org/wiki/Folly) I would remove that option.